ddtrace/tracer/spancontext.go

```diff
@@ -126,7 +126,7 @@ func (c *spanContext) baggageItem(key string) string {
 }
 
 // finish marks this span as finished in the trace.
-func (c *spanContext) finish() { c.trace.finishedOne(c.span) }
+func (c *spanContext) finish(keep bool) { c.trace.finishedOne(c.span, keep) }
```

Comment:
```suggestion
func (c *spanContext) finishAndKeep(keep bool) {
    c.trace.setKeep(keep)
    c.trace.finishedOne(c.span)
}
```
Boolean args are generally considered a smell. I think it's fine to use them, if it's the only one argument, and the name somehow makes it clear what it is. What do you think?